pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,192 @@
+import { ClipboardCopy } from '@patternfly/react-core/dist/esm/components/ClipboardCopy/index.js';
+import { EmptyState, EmptyStateBody } from '@patternfly/react-core/dist/esm/components/EmptyState/index.js';
+import { FormGroup, FormFieldGroup, FormFieldGroupHeader } from '@patternfly/react-core/dist/esm/components/Form/index.js';
+import { Grid } from '@patternfly/react-core/dist/esm/layouts/Grid/index.js';
+import { InputGroup } from '@patternfly/react-core/dist/esm/components/InputGroup/index.js';
+import { MinusIcon } from '@patternfly/react-icons';
+import { TextInput } from '@patternfly/react-core/dist/esm/components/TextInput/index.js';
+
+import { Name, NetworkModal, dialogSave } from "./dialogs-common";
+import { ModelContext } from './model-context';
+import { useDialogs } from 'dialogs.jsx';
+
+import './wireguard.scss';
+
+const _ = cockpit.gettext;
+
+export function WireGuardDialog({ settings }) {
+    const Dialogs = useDialogs();
+    const idPrefix = 'network-wireguard-settings';
+    const model = useContext(ModelContext);
+
+    const [iface, setIface] = useState(settings.connection.interface_name);
+    const [privateKey, setPrivateKey] = useState('');
+    const [publicKey, setPublicKey] = useState('');
+    const [listenPort, setListenPort] = useState('');
+    const [addresses, setAddresses] = useState('');
+    const [dialogError, setDialogError] = useState('');
+    const [peers, setPeers] = useState([]);
+
+    useEffect(() => {
+        async function getPublicKey() {
+            try {
+                const key = await cockpit.script(`echo ${privateKey.trim()} | wg pubkey`, { err: 'message' });
+                setPublicKey(key.trim());
+            } catch (e) {
+                setPublicKey('');
+            }
+        }
+
+        getPublicKey();
+    }, [privateKey]);
+
+    function onSubmit() {
+        function createSettingsObj() {
+            return {
+                connection: {
+                    id: `con-${iface}`,
+                    interface_name: iface,
+                    type: 'wireguard'
+                },
+                wireguard: {
+                    private_key: privateKey.trim(),
+                    listen_port: +listenPort,
+                    peers,
+                },
+                ipv4: {
+                    addresses: [[addresses.split('/')[0], addresses.split('/')[1], '0.0.0.0']], // TODO: fix gateway
+                    method: 'manual'
+                }
+            };
+        }
+
+        dialogSave({
+            model,
+            settings: createSettingsObj(),
+            onClose: Dialogs.close,
+            setDialogError
+        });
+    }
+
+    return (
+        <NetworkModal title={_("WireGuard settings")} onSubmit={onSubmit} dialogError={dialogError} idPrefix={idPrefix}>
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_('Private key')} fieldId={idPrefix + '-private-key-input'}>
```

Comment:
Hmm, this ought to trigger test/static-code. Translatable text must be enclosed in double quotes.

*looks* -- and indeed [it does](https://github.com/cockpit-project/cockpit/actions/runs/5392098900/jobs/9789951294?pr=19024#step:4:653)

I suggest to set up the pre-push check. Search https://github.com/cockpit-project/cockpit/blob/main/HACKING.md#working-on-the-non-web-parts-of-cockpit for "pre-push hook".